src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -281,33 +281,47 @@ private List<(DocumentId, AsyncLazy<DocumentAnalysisResults>)> GetChangedDocumen
             return result;
         }
 
-        private async Task<HashSet<ISymbol>> GetAllAddedSymbols(CancellationToken cancellationToken)
+        private async Task<HashSet<ISymbol>> GetAllAddedSymbolsAsync(Project project, CancellationToken cancellationToken)
         {
-            Analysis[] analyses;
-            lock (_analysesGuard)
+            try
             {
-                analyses = _analyses.Values.ToArray();
-            }
+                Analysis[] analyses;
+                lock (_analysesGuard)
+                {
+                    analyses = _analyses.Values.ToArray();
+                }
 
-            HashSet<ISymbol> addedSymbols = null;
-            foreach (var analysis in analyses)
-            {
-                var results = await analysis.Results.GetValueAsync(cancellationToken).ConfigureAwait(false);
-                foreach (var edit in results.SemanticEdits)
+                HashSet<ISymbol> addedSymbols = null;
+                foreach (var analysis in analyses)
                 {
-                    if (edit.Kind == SemanticEditKind.Insert)
+                    // Only consider analyses for documents that belong the currently analyzed project.
+                    if (analysis.Document.Project == project)
                     {
-                        if (addedSymbols == null)
+                        var results = await analysis.Results.GetValueAsync(cancellationToken).ConfigureAwait(false);
+                        if (!results.HasChangesAndErrors)
```

Comment:
>if (!results.HasChangesAndErrors) [](start = 24, length = 33)

I believe you intentionally just skipping analysis providing errors not making a complete bail out.
Maybe a comment would be helpful.